### PR TITLE
proposal for container level targeting in PodPreset

### PR DESCRIPTION
**Pull request by droot**
This proposal introduces following changes to PodPreset API
  * extends API to support injection of information in init-containers. Currently information is injected only in app containers.
  * Introduces concept of ContainerSelector so that users can specify init-containers/app-containers to inject information in to.

### CODE REVIEW

File: contributors/design-proposals/pod-preset.md [hunk 1]
```diff
@@ -133,6 +133,19 @@ type PodPresetSpec struct {
     // Selector is a label query over a set of resources, in this case pods.
     // Required.
     Selector      unversioned.LabelSelector
+
```
Comment (pwittrock) [inline contributors/design-proposals/pod-preset.md]: Format should be

`InitContainerSelector is ...`

File: contributors/design-proposals/pod-preset.md [hunk 2]
```diff
@@ -147,6 +162,46 @@ type PodPresetSpec struct {
...
+    // Empty or nil list is treated as match NOTHING.
+    MatchNames []string
+
+   // in future matchExpressions support can also be added to enable set expresssions.
+}
+
+// Matches returns list of containers matching selection criterion specified in
+// given container selector cs.
+func (cs *ContainerSelector) Matches(ctrs []v1.Container) []v1.Container {
+	if cs == nil {
```
Comment (duglin) [inline contributors/design-proposals/pod-preset.md]: I think gofmt needs to be run on these files.

File: contributors/design-proposals/pod-preset.md [hunk 3]
```diff
@@ -133,6 +133,21 @@ type PodPresetSpec struct {
...
     Selector      unversioned.LabelSelector
+
+    // InitContainerSelector is a ContainerSelector to select init containers
+    // of a PodSpec for injecting information into.
+    // If not specified, it will default to EmptyContainerSelector which matches
+    // any init-container in the PodSpec.
+    // +optional
+    InitContainerSelector *ContainerSelector
+
+    // AppContainerSelector is a ContainerSelector to select app containers of a
```
Comment (liggitt) [inline contributors/design-proposals/pod-preset.md]: what is an app container?

Comment (derekwaynecarr) [inline contributors/design-proposals/pod-preset.md]: prefer we have InitContainerSelector and ContainerSelector, not introduce App prefix.

Comment (droot) [inline contributors/design-proposals/pod-preset.md]: @liggitt app container refer to regular containers. Realized App prefix is not a good idea :).
So removed the prefix, now AppContainerSelector --> ContainerSelector.

File: contributors/design-proposals/pod-preset.md (hunk 2 above)
Comment (droot) [inline contributors/design-proposals/pod-preset.md]: Fixed the formatting.

**Comment by droot** [comment]
@derekwaynecarr @liggitt Addressed the review comments. PTAL.

**Pull request closed without merging**